packages/sdk/src/bot/teamsBotSsoPrompt.ts

```diff
@@ -284,7 +286,13 @@ export class TeamsBotSsoPrompt extends Dialog {
   private async sendOAuthCardAsync(context: TurnContext): Promise<void> {
     internalLogger.verbose("Send OAuth card to get SSO token");
 
-    const signInResource = this.getSignInResource();
+    const account: TeamsChannelAccount = await TeamsInfo.getMember(
+      context,
+      context.activity.from.id
+    );
+    internalLogger.verbose("Get Teams member account email address: " + account.email);
```

Comment:
Some tenant may not use email as login account, can you see whether there's other property like user principle name?
https://docs.microsoft.com/en-us/azure/active-directory/hybrid/plan-connect-userprincipalname

Reply:
Changed to userPrincipalName.